eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Enact.hs

```diff
@@ -50,19 +52,20 @@ instance EraGov era => STS (ConwayENACT era) where
   initialRules = []
   transitionRules = [enactmentTransition]
 
-enactmentTransition :: forall era. EraPParams era => TransitionRule (ConwayENACT era)
+enactmentTransition :: forall era. EraGov era => TransitionRule (ConwayENACT era)
 enactmentTransition = do
   TRC ((), st, act) <- judgmentContext
+
   case act of
     ParameterChange _prevGovActionId ppup -> pure $ st {ensPParams = newPP}
       where
         newPP = ensPParams st `applyPPUpdates` ppup
     HardForkInitiation _prevGovActionId pv -> pure $ st {ensProtVer = pv}
     TreasuryWithdrawals wdrls -> do
-      let wdrlsAmount = fold wdrls
-          -- TODO: validate NetworkId in all
-          -- We can drop NetworkIds after their validation
-          wdrlsNoNetworkId = Map.mapKeys getRwdCred wdrls
+      expectedNetworkId <- liftSTS $ asks networkId
+      let withdrawals = Map.filterWithKey (\rwrdAcnt _ -> getRwdNetwork rwrdAcnt == expectedNetworkId) wdrls
+          wdrlsAmount = fold withdrawals
+          wdrlsNoNetworkId = Map.mapKeys getRwdCred withdrawals
```

Comment:
NetworkId for the withdrawal account need to be validated upon proposal, i.e. in the `GOV` rule, not during Enactment.

If it is validated upon proposal, then here, when we ENACT this withdrawal action we can simply ignore the NetworkId value